src/librustdoc/clean/simplify.rs

```diff
@@ -24,7 +24,10 @@ use crate::core::DocContext;
 
 crate fn where_clauses(cx: &DocContext<'_>, clauses: Vec<WP>) -> Vec<WP> {
     // First, partition the where clause into its separate components
-    let mut params: BTreeMap<_, (Vec<_>, Vec<_>)> = BTreeMap::new();
+    let mut params: BTreeMap<Symbol, (Vec<_>, Vec<_>)> = BTreeMap::new();
+    // The BTreeMap cannot keep items sorted simply because the `PartialOrd`implementation on
+    // `Symbol` isn't string based but integer based!
```

Comment:
If the goal is to keep these insert-ordered, it seems like the right thing to do is switch to a FxIndexMap here.

I'm not sure why *sorting* the symbols was ever intended; that doesn't seem like it would have kept them in the original order even if they sorted by string (and not integer).

Reply:
Seems like this was done here - https://github.com/rust-lang/rust/commit/421e6fc24d2e5bfc59746d2662531fb22f7e0720 - presumably with the goal of deterministic output. I'm not actually sure that was achieved anyway (since Symbol isn't *really* ordered), but regardless, it seems like an FxIndexMap should do the trick.

Reply:
Nice, thanks!